pkg/distributors/contracts/MultiRewards.sol

```diff
@@ -241,11 +241,20 @@ contract MultiRewards is IMultiRewards, IDistributor, ReentrancyGuard, MultiRewa
         IERC20 pool,
         uint256 amount,
         address receiver
-    ) public nonReentrant updateReward(pool, receiver) {
+    ) public nonReentrant {
+        _stakeFor(pool, amount, msg.sender, receiver);
+    }
+
+    function _stakeFor(
+        IERC20 pool,
+        uint256 amount,
+        address account,
+        address receiver
+    ) internal updateReward(pool, receiver) {
```

Comment:
```suggestion
    ) internal nonReentrant updateReward(pool, receiver) {
```

I'd move the reentrancy guard to here to ensure that we don't forget to add another way to stake and forget to add the guard there.